### PR TITLE
add logs when stuck

### LINKED ISSUE #33: Add logs when stuck
Currently, we detect when the agent gets stuck and abort

It would be great instead to
1. check with the LLM if it is stuck because of a bad test command, or just the same test failure on repeat
2. if ti is test failure on repeat - instruct the LLM to add logs to the code and set an internal flag that this was set

the idea is, with some logs, perhaps this will help the AI see more info and get unstuck just like a human

then, if the flag is on, at the end tell the AI to remove the logs from the code before fully finishing

### PR DESCRIPTION
**Pull request by nexxeln**
closes #33

i took a shot at implementing this. lemme know if there's any changes or improvements needed!

### CODE REVIEW

File: src/helpers/run.ts [hunk 1]
```diff
@@ -140,3 +155,38 @@ export async function runAll(
...
+    messages: [
+      {
+        role: 'system',
+        content:
+          'You are an assistant that helps improve code by adding logs for debugging.',
+      },
+      {
+        role: 'user',
+        content: `Please add detailed logs to the following code to help debug repeated test failures:\n\n${options.priorCode}`,
+      },
```
Comment (steve8708) [inline src/helpers/run.ts:170]: would be great to pass the test failure text in here too so it has more context about where things are stuck

Comment (nexxeln) [reply src/helpers/run.ts:170]: done

Comment (steve8708) [reply src/helpers/run.ts:170]: for an optimal prompt, when we have multiple variables, we should wrap these in xml tags, like

```
content: `Please add detailed logs to the following code to help debug repeated test failures:\n\n<code>${options.priorCode}</code>\n\nThe error you received on that code was:\n\n<error>${options.lastRunError}</error>`,
```

Comment (nexxeln) [reply src/helpers/run.ts:170]: done

**Comment by steve8708** [comment]
awesome, thanks @nexxeln! merging this in and i'll do some testing

**Pull request merged**